src/js/components/home/ProgressSection.js

```diff
@@ -0,0 +1,144 @@
+import React, {PropTypes} from 'react';
+import { Link } from 'react-router';
+import { getProgresses } from '../../actions/progressActions';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import InlineLoader from '../common/InlineLoader';
+
+class ProgressSection extends React.Component {
+  constructor(props) {
+    super(props);
+    this.animateProgess = this.animateProgess.bind(this);
+  }
+
+  componentDidMount() {
+    document.addEventListener('scroll', this.animateProgess);
+    this.props.getProgresses()
+  }
+
+  componentWillUnmount() {
+    document.removeEventListener('scroll', this.animateProgess);
+  }
+
+  getProgressValueAndPercentage(progressData, key, total) {
+    if (progressData) {
+      return {
+        value: progressData[key],
+        percentage: Math.round(progressData[key] / total * 100),
+        total: total
+      };
+    }
+    return { value: 0, percentage: 0, total: total };
+  }
+
+  animateProgess(event) {
+    document.querySelectorAll('.progress-circle').forEach((progressCircle, index) => {
+      const scrollTop = document.body.scrollTop;
+      const progressCircleTop = progressCircle.getBoundingClientRect().top + window.scrollY;
+      const heightToStartOfProgressCircle = progressCircleTop + progressCircle.clientHeight - window.innerHeight;
+
+      if (scrollTop < heightToStartOfProgressCircle) {
+        progressCircle.setAttribute('data-progress', 0);
+      } else {
+        if (progressCircle.getAttribute('data-progress') !== progressCircle.getAttribute('data-goal')) {
+          progressCircle.setAttribute('data-progress', progressCircle.getAttribute('data-goal'));
+        }
+      }
+    }, this);
+  }
+
+  generateProgressCircle(data, goal, color) {
+    let progressStat;
+    const { hasBeenFetched } = this.props.progress;
+    if (!hasBeenFetched) {
+      progressStat = <InlineLoader/>
+    } else {
+      progressStat = (
+        <span>
+          {data.value}/<span>{data.total}</span>
+        </span>
+      );
+    }
```

Comment:
`ProgressStat` feels like it has enough going on to be a component on its own.

Edit. The more appropriate component would be `ProgressCircle`